contracts/PetPark.sol

```diff
@@ -84,7 +84,11 @@ contract PetPark {
             revert("Invalid Age");
         }
 
-        if (u.gender != _gender) {
+        if (
+            u.gender != _gender ||
+            (u.gender == Gender.Female && _gender == Gender.Male) ||
+            (u.gender == Gender.Male && _gender == Gender.Female)
+        ) {
```

Comment:
I'm still not able to pass the female under 40 borrowing cats because it hits this test. Here is my logic for the code:

`u.gender != _gender` - this is the clear incorrect case.
`(u.gender == Gender.Female && _gender == Gender.Male)` - this is an example of another clear case, feels redundant but it shows genders not matching
`(u.gender == Gender.Male && _gender == Gender.Female)` - this is the case I have an issue with. If I am understanding solidity correctly, before an address is added to the map, it will default to male since it is first in the Gender enum. Therefore, if an address calls `borrow` for the first time and is of `Gender.Female`, this will always fail because it won't match the default option on the enum. I need clarification as to how to get around this. I don't know why there isn't a need for a `Gender.None` enum option to avoid this, like there is for `AnimalType`.